components/raftstore/src/store/worker/read.rs

```diff
@@ -263,6 +268,35 @@ impl ReadDelegate {
 
         false
     }
+
+    fn is_safe_to_read<S: Snapshot>(
```

Comment:
It feels odd and unclear to return `Option` for an `is_xxx` method. How about naming it `check_stale_read_safe` and return `Result<(), ReadResponse<S>>`?